src/interactions/brush.js

```diff
@@ -0,0 +1,93 @@
+import {create} from "../context.js";
+import {brush as brusher, brushX as brusherX, brushY as brusherY, union} from "d3";
+import {composeRender} from "../mark.js";
+
+function brushTransform(mode, options) {
+  return {
+    ...options,
+    // Unlike other composed transforms, interactive transforms must be the
+    // outermost render function because they will re-render dynamically in
+    // response to pointer events.
+    render: composeRender(function (index, scales, values, dimensions, context, next) {
+      const {data} = context.getMarkState(this);
+      const {width, height, marginLeft, marginTop, marginRight, marginBottom} = dimensions;
+      context.dispatchValue(data);
+      let viz = next.call(this, [], scales, values, dimensions, context);
+      const {x, y} = scales;
+      const {x: X, x1: X1, x2: X2, y: Y, y1: Y1, y2: Y2} = values;
+
+      const Xl = X1 && X2 ? X1.map((d, i) => Math.min(d, X2[i])) : X;
+      const Xm = X1 && X2 ? X1.map((d, i) => Math.max(d, X2[i])) : X;
+      const Yl = Y1 && Y2 ? Y1.map((d, i) => Math.min(d, Y2[i])) : Y;
+      const Ym = Y1 && Y2 ? Y1.map((d, i) => Math.max(d, Y2[i])) : Y;
+
+      // Use a RAF so we have access to the (facet) transform
+      window.requestAnimationFrame?.(() => {
+        const transform = viz.getAttribute("transform"); // Facet transform
+        // Register all brushes on the mark so they can communicate
+        if (!this.brushes) this.brushes = [];
```

Comment:
This needs to use the per-plot states WeakMap approach that the pointer interaction does. If the same mark/interaction is rendered across multiple plots there should be no shared state (and no memory leak).